apps/meteor/app/lib/server/functions/deleteUser.ts

```diff
@@ -15,19 +16,15 @@ import { LivechatUnitMonitors } from '../../../../ee/app/models/server';
 
 export async function deleteUser(userId: string, confirmRelinquish = false): Promise<void> {
 	const user = Users.findOneById(userId, {
-		fields: { username: 1, avatarOrigin: 1, federation: 1, roles: 1 },
+		fields: { username: 1, avatarOrigin: 1, roles: 1, federated: 1 },
 	});
 
 	if (!user) {
 		return;
 	}
 
-	if (user.federation) {
-		const existingSubscriptions = Subscriptions.find({ 'u._id': user._id }).count();
-
-		if (existingSubscriptions > 0) {
-			throw new Meteor.Error('FEDERATION_Error_user_is_federated_on_rooms');
-		}
+	if (isUserFederated(user)) {
+		throw new Meteor.Error('error-user-is-federated', 'Cannot delete federated user');
```

Comment:
Is this right? Feels like a weird rule.

Once a federated user is created it'll stay on the server forever? what about GDPR?

Reply:
From what we have until now (we don't have anything integrated with Matrix regarding users' deletion) and for the scope of this PR, yes. But for sure this is not correct, we'll handle the user's deletion in this [task](https://rocketchat.atlassian.net/browse/FED-74). Thanks for the heads up about GDPR.